internal/app/app.go

```diff
@@ -133,26 +134,33 @@ func (hs *HttpService) ToMachineService() *api.MachineService {
 		}},
 		Concurrency: concurrency,
 	}
-
 }
 
-func (s *Service) ToMachineService() *api.MachineService {
+func (s *Service) toMachineService() *api.MachineService {
+	checks := make([]api.Check, len(s.TCPChecks)+len(s.HTTPChecks))
```

Comment:
I'm pretty sure this makes a slice with `len(s.TCPChecks)+len(s.HTTPChecks)` items all set to nil. the appends add to that list without overwriting any of the nils. 

```suggestion
	checks := make([]api.Check, 0)
```

Reply:
I'm playing around with this, and will push an update in a sec.